modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/actions/ActionForm.tsx

```diff
@@ -329,35 +389,93 @@ const ActionForm = ({
 							className="align-items-center d-flex justify-content-end"
 							size={4}
 						>
-							<ClayIcon
-								className="mr-4"
-								symbol={actionData.iconSymbol}
-							/>
-
 							<ClayForm.Group>
 								<label htmlFor={iconFormElementId}>
 									{Liferay.Language.get('icon')}
 								</label>
 
-								<ClaySelectWithOption
-									id={iconFormElementId}
-									onChange={(event) =>
-										setActionData({
-											...actionData,
-											iconSymbol: event.target.value,
-										})
-									}
-									options={[
-										{
-											label: Liferay.Language.get(
-												'select'
-											),
-											value: '',
-										},
-										...availableIconSymbols,
-									]}
-									value={actionData.iconSymbol || ''}
-								/>
+								<ClayInput.Group>
+									<ClayInput.GroupItem prepend shrink>
+										<ClayInput.GroupText>
+											<ClayIcon
+												spritemap={spritemap}
+												symbol={actionData.iconSymbol}
+											/>
+										</ClayInput.GroupText>
+									</ClayInput.GroupItem>
+
+									<ClayInput.GroupItem append>
+										<ClayInput
+											placeholder={Liferay.Language.get(
+												'no-icon-selected'
+											)}
+											type="text"
+											value={actionData.iconSymbol}
+										/>
+									</ClayInput.GroupItem>
+
+									{actionData.iconSymbol === '' ? (
+										<ClayButtonWithIcon
+											aria-label={Liferay.Language.get(
+												'add-icon'
+											)}
+											className="ml-2"
+											displayType="secondary"
+											id={iconFormElementId}
+											onClick={() =>
+												openModal({
+													bodyComponent: ModalBody,
+													containerProps: {
+														className: 'icons-list',
+													},
+													size: 'lg',
+													title: Liferay.Language.get(
+														'select-an-icon'
+													),
+												})
+											}
+											symbol="plus"
+										/>
+									) : (
+										<>
+											<ClayButtonWithIcon
+												aria-label={Liferay.Language.get(
+													'change-icon'
+												)}
+												className="ml-2"
+												displayType="secondary"
+												onClick={() =>
+													openModal({
+														bodyComponent: ModalBody,
+														containerProps: {
+															className:
+																'icons-list',
+														},
+														size: 'lg',
+														title: Liferay.Language.get(
+															'select-an-icon'
+														),
+													})
+												}
+												symbol="change"
+											/>
```

Comment:
There is a lot of code duplication here. Could we simplify this by inline checks on `aria-label` and `symbol`=

Reply:
I think that way do more for behavior global understanding, but I've [commit a refactor](https://github.com/liferay-frontend/liferay-portal/pull/3772/commits/e6f9abdf6f475e33c2b0e89060ec5ae894ea2394) to avoid code duplication, so we can compare and see which works the most for us.